src/vs/workbench/services/search/common/fileSearchManager.ts

```diff
@@ -333,17 +333,17 @@ export class FileSearchManager {
 	}
 
 	clearCache(cacheKey: string): void {
-		const sessionTokenSource = this.getSessionTokenSource(cacheKey);
-		sessionTokenSource?.cancel();
+		// with no reference to this, it will be removed from WeakMaps
+		this.sessions.delete(cacheKey);
```

Comment:
Is `this.sessions` supposed to be a WeakMap? Looks like a normal map

Reply:
It would be a weakmap on the extension side, but on our side, we explicitly reference it. When we delete our reference, it should delete on the extension side's weakmap. @jrieken @mjbvz is this the correct way of going about it?

Reply:
I get it. Btw, the dts still describes it as a token: `When the token is cancelled, the session is complete and the cache can be cleared.`

The change looks fine though.

Reply:
On `FileSearchProviderOptions#session` you should probably mention something about the extension not retaining any references to the session object, or it uses a WeakMap, or we just have a good example.

Reply:
fixed in https://github.com/microsoft/vscode/pull/226900